tests/php/test-fieldmanager-checkboxes-field.php

```diff
@@ -0,0 +1,172 @@
+		$fm = new Fieldmanager_Checkboxes( array(
+			'name' => 'base_field',
+			'options' => array( 'one', 'two', 'three' ),
+		) );
+
+		$html = $this->_get_html_for( $fm, array( 'two' ) );
+
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'one' ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'two', true ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'three' ), $html );
+	}
+
+	public function test_associative_render() {
+		$fm = new Fieldmanager_Checkboxes( array(
+			'name' => 'base_field',
+			'options' => array( 1 => 'one', 2 => 'two', 3 => 'three' ),
+		) );
+
+		$html = $this->_get_html_for( $fm );
+
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', array( 1 => 'one' ) ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', array( 2 => 'two' ) ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', array( 3 => 'three' ) ), $html );
+
+		$html = $this->_get_html_for( $fm, array( 2 ) );
+
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', array( 1 => 'one' ) ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', array( 2 => 'two' ), true ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', array( 3 => 'three' ) ), $html );
+	}
+
+	public function test_default_value() {
+		$fm = new Fieldmanager_Checkboxes( array(
+			'name' => 'base_field',
+			'options' => array( 'one', 'two', 'three' ),
+			'default_value' => 'two',
+		) );
+
+		$html = $this->_get_html_for( $fm );
+
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'one' ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'two', true ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'three' ), $html );
+	}
+
+	public function test_datasource() {
+		$fm = new Fieldmanager_Checkboxes( array(
+			'name' => 'base_field',
+			'datasource' => $this->custom_datasource,
+		) );
+
+		$html = $this->_get_html_for( $fm );
+
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'January' ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'February' ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'December' ), $html );
+	}
+
+	public function test_datasource_default_value() {
+		$fm = new Fieldmanager_Checkboxes( array(
+			'name' => 'base_field',
+			'default_value' => 'February',
+			'datasource' => $this->custom_datasource,
+		) );
+
+		$html = $this->_get_html_for( $fm );
+
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'January' ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'February', true ), $html );
+		$this->assertRegExp( $this->_get_input_field_regex( 'base_field', 'December' ), $html );
+	}
+
+	public function test_datasource_save() {
```

Comment:
This test gets flagged as risky, which I don't understand much yet, @mboynes can you maybe take a look and teach me a bit? :angel: